esp/esdllib/esdl_script.cpp

```diff
@@ -396,22 +396,37 @@ class CEsdlTransformOperationHttpPostXml : public CEsdlTransformOperationBase
         {
             StringBuffer status;
             StringBuffer response;
-            httpclient->sendRequest(headers, "POST", "text/xml", content, response, status);
+            StringBuffer errmsg;
+            HttpClientErrCode code = HttpClientErrCode::OK;
+            Owned<IHttpMessage> resp = httpclient->sendRequestEx("POST", "text/xml", content, code, errmsg, headers);
             targetContext->ensureSetValue("@status", status.str(), true);
-            size32_t len = response.trim().length();
-            if (!len)
-            {
+
+            StringBuffer err;
+            err.append((int) code);
+            targetContext->ensureSetValue("@error-code", err.str(), true);
+            if (code != HttpClientErrCode::OK)
+                throw MakeStringException(ESDL_SCRIPT_Error, "ESDL Script error sending request in http-post-xml %s url(%s)", m_traceName.str(), url);
+
+            resp->getStatus(status);
+            targetContext->ensureSetValue("@status", status.str(), true);
+
+            resp->getContent(response);
+            if (!response.trim().length())
                 throw MakeStringException(ESDL_SCRIPT_Error, "ESDL Script empty result calling http-post-xml %s url(%s)", m_traceName.str(), url);
-            }
-            else if (*response.str()=='<')
+
+            StringBuffer contentType;
+            resp->getContentType(contentType);
+            targetContext->ensureSetValue("@content-type", contentType.str(), true);
+            if (strnicmp("text/xml", contentType.str(), 8)==0 || strnicmp("application/xml", contentType.str(), 15) ==0)
             {
                 CXpathContextLocation content_location(targetContext);
                 targetContext->addElementToLocation("content");
                 targetContext->addXmlContent(response.str());
             }
             else
```

Comment:
Curious if, with the availability of content-type, it might be beneficial to recognize JSON (or anything we can load into a PTree) then convert to XML. Add the text so somebody expecting the actual response can get it, but do the conversion automatically so it's ready to use.

On the other hand, if any non-XML content type is considered an error, should we make sure the request asks for XML results?

Reply:
I'll ask for xml.  I named the function with xml in the name to put off the complexity of conversion.  If I were already loading into a ptree it would be different.  Later I can probably create something that parses json directly into libxml.  I don't like the ptree as an intermediary here.